op-chain-ops/cmd/celo-migrate/state.go

```diff
@@ -264,9 +267,11 @@ func applyAllocsToState(db *state.StateDB, genesis *core.Genesis, config *params
 						log.Info("Account already exists with different code and is whitelisted, overwriting...", "address", k)
 					} else {
 						log.Warn("Account already exists with different code and is not whitelisted, overwriting...", "address", k, "oldCode", db.GetCode(k), "newCode", v.Code)
+						return fmt.Errorf("account already exists with different code and is not whitelisted: %s", k.Hex())
 					}
 				} else {
 					log.Warn("Account already exists with different code and no whitelist exists", "address", k, "oldCode", db.GetCode(k), "newCode", v.Code)
+					return fmt.Errorf("account already exists with different code and no whitelist exists: %s", k.Hex())
 				}
 
 				overwriteCounter++
```

Comment:
Currently this function is just checking whether the code matches or not. I think I would be inclined to just fail if there is an existing account. I can't imagine a situation where we are doing this deploy and we find an account that somehow has the same code that we are already deploying.

Reply:
> So either if there's a clash either it's whitelisted, or we fail.

How is that different from what the code currently does?

Reply:
Currently if there is a clash but the code is equal, we'll overwrite the nonce and update the balance.

Reply:
Ok, incorporated a couple of changes:
- Check that the copied account has a balance of zero. otherwise we'd change the total supply of celo.
- Check if we'd touch an EO account with non-zero nonce, if so bail out. Thinking here is that we should not overwrite EOAs.
- If we overwrite and smart contract account, copy the nonce. Contracts have a nonce which is increased when they deploy new contracts, so we want to keep the nonce in that case.